File: csrc/334.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        if (nums.size() < 3) return false;
        int minValue = INT_MAX;
        int secondValue = INT_MAX;
        for (int i = 0; i < nums.size(); i++) {
            if (minValue >= nums[i]) minValue = nums[i];
            else if (secondValue >= nums[i]) secondValue = nums[i];
            else return true;
        }
        return false;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {1, 2, 3, 4, 5};
    cout << solution.increasingTriplet(nums);
    return 0;
}

File: pysrc/334.py
import sys
from typing import List


class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        small = sys.maxsize
        mid = sys.maxsize
        for num in nums:
            if num < small:
                small = num
            elif num > small and num < mid:
                mid = num
            elif num > mid:
                return True
        return False


if __name__ == '__main__':
    solution = Solution()
    print(solution.increasingTriplet([1, 2, 3, 4, 5]) == True)
    print(solution.increasingTriplet([5, 4, 3, 2, 1]) == False)
    print(solution.increasingTriplet([2, 1, 5, 0, 4, 6]) == True)
    print(solution.increasingTriplet([1, 0, 2, 0, 3]) == True)
